internal/controllers/synthetic_devices_controller.go

```diff
@@ -40,7 +40,11 @@ type SyntheticDevicesController struct {
 	deviceDefSvc   services.DeviceDefinitionService
 	walletSvc      services.SyntheticWalletInstanceService
 	registryClient registry.Client
-	teslaOracle    pb_oracle.TeslaOracleClient
+	teslaOracle    TeslaOracleClient
+}
+
+type TeslaOracleClient interface {
```

Comment:
Agree, better design approach

Reply:
Eh it's mostly so that I can mock it in the test.